### PR TITLE
Kokkos charmmfsw pair and dihedral styles

### PR DESCRIPTION
**Pull request by alphataubio**
**Summary**

add pair/lj/charmmfsw/coul/long/kk and dihedral/charmmfsw/kk 

**Related Issue(s)**

https://matsci.org/t/charmmfsw-pair-style-and-gpu-acceleration/39482

**Author(s)**

Mitch Murphy, alphataubio.com, alphataubio@gmail.com

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

n/a

**Implementation Notes**

*** DRAFT VERSION 1 (lots of comments to be removed just before merge) ***
 
(1) first draft version of PairLJCharmmfswCoulLongKokkos exactly same as PairLJCharmmCoulLongKokkos but with new class name. compiles without errors or warnings
 
method: track changes from serial kspace pair_lj_charmm_coul_long to pair_lj_charmmfsw_coul_long and apply to PairLJCharmmfswCoulLongKokkos
 
% diff pair_lj_charmm_coul_long.h pair_lj_charmmfsw_coul_long.h

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [x] One or more example input decks are included

**Further Information, Files, and Links**

My test input is a protein-dna simulation (1HVN), the smallest number of atoms available on PDB with at least one protein, at least one dna, and no modified engineered residues. 1HVN lammps input scripts created by charmm-gui.org:

[1hvn-lammps-charmm.zip](https://github.com/lammps/lammps/files/13808846/1hvn-lammps-charmm.zip)

[1hvn-lammps-charmmfsw.zip](https://github.com/lammps/lammps/files/13808848/1hvn-lammps-charmmfsw.zip)

### CODE REVIEW

**Comment by stanmoore1** [comment]
@alphataubio this now gives correct numerics on the GPU for my test, and does not hang.

File: examples/charmm/1hvn/restraints/dihe.restraint
Comment (stanmoore1) [inline examples/charmm/1hvn/restraints/dihe.restraint:1]: Remove empty file, or put in a comment why it is needed.

File: examples/charmm/1hvn/sysinfo.dat [hunk 1]
```diff
@@ -0,0 +1 @@
+{"membres": {"head": [], "head_upper": [], "head_lower": []}}
```
Comment (stanmoore1) [inline examples/charmm/1hvn/sysinfo.dat]: Is this file actually needed? If so, it needs a "newline" at the end of the file, github is flagging it.

**Review by stanmoore1 — CHANGES_REQUESTED** [review]
@alphataubio overall this is almost ready to merge, just a few minor change requests.

**Review by stanmoore1 — APPROVED** [review]

**Review by ndtrung81 — APPROVED** [review]

**Review by sjplimp — APPROVED** [review]

**Comment by alphataubio** [comment]
for documentation purposes and reproducibility later, im attaching to this PR mathematica notebook used to create plots. had to use zip because .nb files not allowed on github

[Howto_bioFF.nb.zip](https://github.com/lammps/lammps/files/13905534/Howto_bioFF.nb.zip)

**Pull request merged**